### PR TITLE
chore: move autocomplete to modules/

### PR DESCRIPTION
**Pull request by yohanboniface**

### CODE REVIEW

File: umap/static/umap/js/modules/autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
+import { DomUtil, DomEvent, setOptions } from '../../vendors/leaflet/leaflet-src.esm.js'
+import { translate } from './i18n.js'
+import { ServerRequest } from './request.js'
+
+export class BaseAutocomplete {
+  constructor(el, options) {
+    this.el = el
+    this.options = {
+      placeholder: translate('Start typing...'),
+      emptyMessage: 'No result',
```
Comment (davidbgk) [inline umap/static/umap/js/modules/autocomplete.js:10]: No translate for that one?

File: umap/static/umap/js/modules/autocomplete.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  }
+
+  hide() {
+    this.clear()
+    this.container.style.display = 'none'
+    this.input.value = ''
+  }
+
+  setChoice(choice) {
+    console.log(choice)
```
Comment (davidbgk) [inline umap/static/umap/js/modules/autocomplete.js:145]: Oups :)

File: umap/static/umap/js/modules/autocomplete.js [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      this.current = result
+      this.highlight()
+    })
+    DomEvent.on(el, 'mousedown', () => this.setChoice())
+    return result
+  }
+
+  resultToIndex(result) {
+    let out = null
+    this.results.forEach((item, index) => {
```
Comment (davidbgk) [inline umap/static/umap/js/modules/autocomplete.js:178]: Using `this.results.filter()` is probably more appropriate to retrieve an element from the list?

File: umap/static/umap/js/modules/autocomplete.js [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    }
+  }
+
+  async search() {
+    let val = this.input.value
+    if (val.length < this.options.minChar) {
+      this.clear()
+      return
+    }
+    if (`${val}` === `${this.cache}`) return
```
Comment (davidbgk) [inline umap/static/umap/js/modules/autocomplete.js:247]: Weird comparison, why don't we compare directly the variables?

File: umap/static/umap/js/modules/autocomplete.js (hunk 3 above)
Comment (yohanboniface) [reply umap/static/umap/js/modules/autocomplete.js]: `filters` will return an array, but `findIndex` seems to do exactly what I need, thanks for the hint!

File: umap/static/umap/js/modules/autocomplete.js (hunk 4 above)
Comment (yohanboniface) [reply umap/static/umap/js/modules/autocomplete.js]: No idea, I removed. I'm not totally why this check if for, actually. So let's fix and see :p

**Review by davidbgk — APPROVED** [review]

**Pull request merged**